compiler/rustc_interface/src/util.rs

```diff
@@ -346,7 +345,7 @@ pub fn rustc_path<'a>() -> Option<&'a Path> {
 }
 
 fn get_rustc_path_inner(bin_path: &str) -> Option<PathBuf> {
-    sysroot_candidates().iter().find_map(|sysroot| {
+    Some(filesearch::get_or_default_sysroot()).iter().find_map(|sysroot| {
```

Comment:
The whole iteration and `find_map` can be eliminated now that there's just a single element.